libphobos/m4/autoconf.m4

```diff
@@ -0,0 +1,134 @@
+#
+# Minimal autoconf support for the D language.
+# Adapted from the Go language support files.
```

Comment:
Probably needs some more work before upstreaming, but that's the plan. OTOH it would take way too long till we can use the upstream implementation, so we'll need a local implementation for now anyway.